src/Apps/Artwork/Components/RelatedWorks/index.tsx

```diff
@@ -0,0 +1,127 @@
+import React from "react"
+import { createFragmentContainer, graphql } from "react-relay"
+import { useTracking } from "react-tracking"
+
+import { ContextModule } from "@artsy/cohesion"
+import * as DeprecatedSchema from "@artsy/cohesion/dist/DeprecatedSchema"
+import { Box, Skeleton, Spacer } from "@artsy/palette"
+
+import { useSystemContext } from "System"
+import { SystemQueryRenderer } from "System/Relay/SystemQueryRenderer"
+import {
+  Header,
+  HeaderPlaceholder,
+} from "Apps/Artwork/Components/OtherWorks/Header"
+import ArtworkGrid, { ArtworkGridPlaceholder } from "Components/ArtworkGrid"
+import { RelatedWorksQuery } from "__generated__/RelatedWorksQuery.graphql"
+import { RelatedWorks_artwork$data } from "__generated__/RelatedWorks_artwork.graphql"
+
+interface RelatedWorksProps {
+  artwork: RelatedWorks_artwork$data
+}
+
+export const RelatedWorks: React.FC<RelatedWorksProps> = ({ artwork }) => {
+  const { trackEvent } = useTracking()
+
+  const artworksConnection = artwork.layer?.artworksConnection
+
+  if (!artworksConnection?.edges?.length) {
+    return null
+  }
+
+  return (
+    <Box data-testid={ContextModule.relatedWorksRail}>
+      <Header title="Related works" />
+
+      <Spacer mt={4} />
+
+      <ArtworkGrid
+        contextModule={ContextModule.relatedWorksRail}
+        artworks={artworksConnection}
+        columnCount={[2, 3, 4, 4]}
+        onBrickClick={() =>
+          trackEvent({
+            type: DeprecatedSchema.Type.ArtworkBrick,
+            action_type: DeprecatedSchema.ActionType.Click,
+            context_module: DeprecatedSchema.ContextModule.RelatedWorks,
+          })
+        }
+      />
+    </Box>
+  )
+}
+
+const PLACEHOLDER = (
+  <Skeleton>
+    <Box>
+      <HeaderPlaceholder title="Related works" />
+
+      <Spacer mt={4} />
+
+      <ArtworkGridPlaceholder columnCount={[2, 3, 4, 4]} />
+    </Box>
+  </Skeleton>
+)
+
+export const RelatedWorksFragmentContainer = createFragmentContainer(
+  RelatedWorks,
+  {
+    artwork: graphql`
+      fragment RelatedWorks_artwork on Artwork {
+        slug
+        title
+        layer(id: "main") {
+          name
+          artworksConnection(first: 8) {
+            ...ArtworkGrid_artworks
+            # Used to check for content
+            edges {
+              node {
```

Comment:
**suggestion / non-blocking**: If we're arbitrarily fetching `node.slug` so we can check that we have edges to render, we could use `__typename` instead.

```
edges {
  __typename
}
```

Reply:
Another alternative, most/all connections offer a `totalCount` connection-level field that might be useful too!

Reply:
Good thoughts! I originally cargo-culted this snippet [from the old implementation](https://github.com/artsy/force/blob/931716dc511598174592480b178f93301cdcbbbb/src/Apps/Artwork/Components/OtherWorks/RelatedWorksArtworkGrid.tsx#L144-L149), but found that `artwork.layer.artworksConnection` didn't support `totalCount` and that requesting only `__typename` didn't change the upstream fetching behavior. So I'm gonna ahead and give this a merge.